frontend/src/basic/OrderPage/index.js

```diff
@@ -109,6 +110,7 @@ class OrderPage extends Component {
       currencyCode: this.getCurrencyCode(newStateVars.currency),
       penalty: newStateVars.penalty, // in case penalty time has finished, it goes back to null
       invoice_expired: newStateVars.invoice_expired, // in case invoice had expired, it goes back to null when it is valid again
+      chat_offset: this.state.chat_offset + newStateVars?.chat?.messages.length,
```

Comment:
This grows the offset exponentially?

Chat offset must be the index of the last message received. 

It is 0 when the component is mount, then it updates to the index of the newest message in the state. After making the call `/api/order ... offset=...` the return contains all new messages. Those have to be merged with the existing messages. That's it ! :D 

We do not need extra API call for turlte mode. All we need for turtle mode is supply an extra param to `/api/order` (the offset).

Reply:
The component is not working. It seems the `offset` param is given to `/api/order` with a NAN value so the backend erros out with 500. Ideally we do not want to supply the `offset` param if we have not activated the turtle mode.

Reply:
this is related to what I said on the previous message.

The NaN error has nothing to do, already resolved

Reply:
> Ideally we do not want to supply the offset param if we have not activated the turtle mode.

Then we'll have to move up the turtle mode boolean one component up, to `OrderPage`, I was trying to avoid this, this is why I always include it

Reply:
> Then we'll have to move up the turtle mode boolean one component up, to `OrderPage`, I was trying to avoid this, this is why I always include it

Yes, it has to be at the same level as the `/api/order` call is. It might look complicated now: but it will be much cleaner once we redo TradeBox and OrderPage.